### PR TITLE
Fix build warnings on other platforms

### PR DESCRIPTION
**Pull request by tinmanjuggernaut**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 *.o
 *.autosave
 *.bc
+*.d
```
Comment (tinmanjuggernaut) [inline .gitignore]: iphone

File: meshers/blocky/voxel_library.cpp [hunk 2]
```diff
@@ -269,7 +269,7 @@ void VoxelLibrary::generate_side_culling_matrix() {
 
 			CRASH_COND(pattern == nullptr);
 
-			if (full_side_pattern_index == -1 && bitmap.all()) {
+			if (full_side_pattern_index == 0xFFFFFFFF && bitmap.all()) {
```
Comment (tinmanjuggernaut) [inline meshers/blocky/voxel_library.cpp]: The warning is here. It doesn't compare about assigning -1, only comparing uint vs -1.

File: SCsub [hunk 3]
```diff
@@ -22,6 +22,9 @@ files = [
 for f in files:
 	env_voxel.add_source_files(env.modules_sources, f)
 
+if env['platform'] == 'osx' or env['platform'] == 'android':
+    env_voxel.Append(CXXFLAGS=['-Wno-inconsistent-missing-override'])
```
Comment (Zylann) [inline SCsub]: Add comment explaining it's because Godot's codebase is old

File: meshers/blocky/voxel_library.cpp [hunk 4]
```diff
@@ -181,7 +181,7 @@ void VoxelLibrary::generate_side_culling_matrix() {
 	};
 
 	std::vector<Pattern> patterns;
-	uint32_t full_side_pattern_index = -1;
+	uint32_t full_side_pattern_index = 0xFFFFFFFF;
```
Comment (Zylann) [inline meshers/blocky/voxel_library.cpp]: Could put that `0xffffffff` in a `NULL_INDEX` constant?

**Comment by tinmanjuggernaut** [comment]
Updates committed.

File: meshers/blocky/voxel_library.h [hunk 5]
```diff
@@ -5,6 +5,8 @@
 #include "voxel.h"
 #include <core/resource.h>
 
+#define NULL_INDEX 0xFFFFFFFF
```
Comment (Zylann) [inline meshers/blocky/voxel_library.h:8]: Use a `static const uint32_t`, not a macro

Comment (tinmanjuggernaut) [reply meshers/blocky/voxel_library.h]: Updated. Why is this way better?

**Comment by Zylann** [comment]
Thanks!

**Pull request merged**